gbpservice/nfp/configurator/agents/loadbalancer_v1.py

```diff
@@ -0,0 +1,655 @@
+        health_monitor = data['health_monitor']
+        pool_id = data['pool_id']
+        driver = self._get_driver(pool_id)
+        assoc_id = {'pool_id': pool_id,
+                    'monitor_id': health_monitor['id']}
+        try:
+            driver.update_pool_health_monitor(old_health_monitor,
+                                              health_monitor,
+                                              pool_id,
+                                              context)
+        except Exception:
+            self._handle_failed_driver_call(
+             'update', 'health_monitor', assoc_id, driver.get_name())
+        else:
+            self.plugin_rpc.update_status(
+                'health_monitor', assoc_id, lb_constants.ACTIVE)
+
+    def _delete_pool_health_monitor(self, ev):
+        data = ev.data
+        context = data['context']
+        health_monitor = data['health_monitor']
+        pool_id = data['pool_id']
+        driver = self._get_driver(pool_id)
+        assoc_id = {'pool_id': pool_id,
+                    'monitor_id': health_monitor['id']}
+        try:
+            driver.delete_pool_health_monitor(health_monitor, pool_id, context)
+        except Exception:
+            LOG.warn(_("Failed to delete pool health monitor."
+                       " assoc_id: %s"), assoc_id)
+
+    def _agent_updated(self, ev):
+        """ TODO:(pritam): Support """
+        return None
+
+    def _handle_failed_driver_call(self, operation, obj_type, obj_id, driver):
+        LOG.error("Failed operation=%s,for obj_type=%s,obj_id=%s,"
+                  "driver=%s " % (operation, obj_type, obj_id, driver))
+        self.plugin_rpc.update_status(obj_type, obj_id, lb_constants.ERROR)
+
+    def _collect_stats(self, ev):
+        self._sc.poll_event(ev)
+
+    @nfp_poll.poll_event_desc(event='COLLECT_STATS', spacing=60)
+    def collect_stats(self, ev):
+        for pool_id, driver_name in LBaaSEventHandler.instance_mapping.items():
+            driver = self.drivers[driver_name]
+            try:
+                stats = driver.get_stats(pool_id)
+                if stats:
+                    self.plugin_rpc.update_pool_stats(pool_id, stats)
+            except Exception:
+                LOG.exception(_('Error updating statistics on pool %s'),
+                              pool_id)
+
+
+def events_init(sc, drivers, rpcmgr, nqueue):
+    """Registers events with core service controller.
+
+    All the events will come to handle_event method of class instance
+    registered in 'handler' field.
+
+    :param drivers: Driver instances registered with the service agent
+    :param rpcmgr: Instance to receive all the RPC messages from configurator
+    module.
+
+    Returns: None
+
+    """
+    evs = [
+        # Events for LBaaS standard RPCs coming from LBaaS Plugin
+        Event(id='CREATE_POOL', handler=LBaaSEventHandler(
+                                                sc, drivers, rpcmgr, nqueue)),
```

Comment:
Better to have a table with all the events and loop to register. Code will be minimal.